proxy/broker.js

```diff
@@ -48,17 +48,21 @@ class Broker {
         }
         switch (xhr.status) {
           case Broker.STATUS.OK:
-            return fulfill(xhr.responseText); // Should contain offer.
-          case Broker.STATUS.GATEWAY_TIMEOUT:
-            return reject(Broker.MESSAGE.TIMEOUT);
+            var response = JSON.parse(xhr.responseText);
+            if (response.Status == "client match") {
```

Comment:
Nitpick: Would a constant like `STATUS_CLIENT_MATCH` (or whatever is idiomatic to JavaScript) be easier to maintain than a string literal? The same question applies to the string literals below.